packages/cli/src/commands/generate/scaffold/scaffold.js

```diff
@@ -416,25 +422,33 @@ const componentFiles = async (
           : null
       }
 
+      const isEnum = column.kind === 'enum'
+      const isList = column.isList
+      const enumType = isEnum && isList ? 'EnumList' : 'Enum'
+      const metadataKey = isEnum ? enumType : column.type
+
       return {
         ...column,
         label: humanize(column.name),
-        component:
-          componentMetadata[column.type]?.componentName ||
-          componentMetadata.default.componentName,
-        defaultProp:
-          componentMetadata[column.type]?.defaultProp ||
-          componentMetadata.default.defaultProp,
-        deserializeFunction:
-          componentMetadata[column.type]?.deserializeFunction ||
-          componentMetadata.default.deserializeFunction,
+        component: isEnum
+          ? componentMetadata[metadataKey]?.componentName
+          : componentMetadata.default.componentName,
```

Comment:
Having the `isEnum` check here looks wrong. Basically now the code will never use `componentMetadata[column.type].componentName` because `metadataKey` will never be `column.type` when `isEnum` is `true`

```suggestion
        component: 
          componentMetadata[metadataKey]?.componentName
          componentMetadata.default.componentName,
```

Reply:
Yes, you're right. 

```ts
        component:
          componentMetadata[metadataKey]?.componentName ||
          componentMetadata.default.componentName,
        defaultProp:
          componentMetadata[metadataKey]?.defaultProp ||
          componentMetadata.default.defaultProp,
        deserializeFunction:
          componentMetadata[metadataKey]?.deserializeFunction ||
          componentMetadata.default.deserializeFunction,
        validation,
        listDisplayFunction:
          componentMetadata[metadataKey]?.listDisplayFunction ||
          componentMetadata.default.listDisplayFunction,
        displayFunction:
          componentMetadata[metadataKey]?.displayFunction ||
          componentMetadata.default.displayFunction,
        values: column.enumValues || [],
```


Cleaned up.